RNGNewAuraNotifier/Core/Config/AppConfig.cs

```diff
@@ -7,120 +7,132 @@ namespace RNGNewAuraNotifier.Core.Config;
 /// </summary>
 internal class AppConfig
 {
-    /// <summary>
-    /// 設定ファイルのパス
-    /// </summary>
-    /// <remarks>アプリケーションの実行ディレクトリに config.json という名前で保存される</remarks>
-    /// <example>config.json</example>
-    private static readonly string _configFilePath = Path.Combine(Environment.CurrentDirectory, "config.json");
+    private const string PathFileName = "config.path";
+    private const string ConfigFileName = "config.json";
+
+    private static string _configDir = GetConfigDirectoryPath();
+    private static ConfigData _instance = new();
+    private static readonly Lock _lock = new();
+    private static bool _isLoaded = false;
 
     /// <summary>
-    /// 設定データ
+    /// 設定が再読み込みされたときに発生するイベント
     /// </summary>
-    private static ConfigData _config = new();
+    public static event Action<ConfigData>? ConfigReloaded;
 
     /// <summary>
-    /// JSONのシリアルオプション
+    /// config.jsonの保存先ディレクトリを取得します。
     /// </summary>
-    private static readonly JsonSerializerOptions _jsonSerializerOptions = new()
+    /// <returns>config.jsonの保存先パス</returns>
+    public static string GetConfigDirectoryPath()
     {
-        WriteIndented = true,
-    };
+        var filePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, PathFileName);
 
-    /// <summary>
-    /// ファイル読み書き時のロックオブジェクト
-    /// </summary>
-    private static readonly Lock _lock = new();
+        if (File.Exists(filePath))
+        {
+            var path = File.ReadAllText(filePath).Trim();
+            return string.IsNullOrEmpty(path) ? AppConstants.ApplicationConfigDirectory : path;
+        }
+        else
+        {
+            // デフォルトの設定ディレクトリを返す
+            return AppConstants.ApplicationConfigDirectory;
+        }
+    }
 
     /// <summary>
-    /// 静的コンストラクタ。設定ファイルを読み込む
+    /// 設定ディレクトリのパスを保存します。
     /// </summary>
-    static AppConfig() => Load();
+    /// <param name="configPath">config.jsonのパス</param>
+    public static void SaveConfigDirectoryPath(string configPath)
+    {
+        var appDir = AppDomain.CurrentDomain.BaseDirectory;
+        var filePath = Path.Combine(appDir, PathFileName);
+
+        if (string.Equals(configPath, AppConstants.ApplicationConfigDirectory, StringComparison.OrdinalIgnoreCase))
+        {
+            // 規定値の場合は config.path を削除
+            if (File.Exists(filePath))
+            {
+                File.Delete(filePath);
+            }
+        }
+        else
+        {
+            // 規定値以外の場合は config.path を作成
+            File.WriteAllText(filePath, configPath);
+        }
```

Comment:
すでに決まっている、なので既定値が正しい？

```suggestion
        if (string.Equals(configPath, AppConstants.ApplicationConfigDirectory, StringComparison.OrdinalIgnoreCase))
        {
            // 既定値の場合は config.path を削除
            if (File.Exists(filePath))
            {
                File.Delete(filePath);
            }
        }
        else
        {
            // 既定値以外の場合は config.path を作成
            File.WriteAllText(filePath, configPath);
        }
```